### ISSUE TITLE
Question regarding Model Training 

### ISSUE THREAD

**Issue by PhucNDA** [CVRP-SOLE/SOLE#10]
Hello,

Thank you for releasing such impressive work!

I have a question regarding the model’s training process. Was the model trained entirely from scratch, or was it fine-tuned from the pretrained Mask3D for ScanNet200? In particular, I noticed in [this section](https://github.com/CVRP-SOLE/SOLE/blob/main/models/sole.py#L83) that the model is loaded from a pretrained backbone. Could you clarify if this pretrained backbone was trained in a fully-supervised manner? If yes, have you tried trained it from scratch, how was the results?

Thank you, and I look forward to your response.

**Comment by 0nandon** [comment]
Nice to meet you @PhucNDA, and read your great work, Open3DIS!

1. We first train Mask3D with only using mask annotations and subsequently extract the backbone weight.
2. We use this backbone weight in the training of SOLE with making it frozen. Basically, only mask decoder and other projection layers are trained during the main training process.

The model can be also trained from the scratch. However, we empirically find that using the pertained backbone can facilitate the convergence and lead to better performance. Unfortunately, we didn't leave the result for that since it was done long time ago. But you are welcome to try by yourself!

Best regards, 

Seungjun Lee.

**Comment by PhucNDA** [comment]
Hi @0nandon,

Thank you for your response—it’s great to hear from you!

From my understanding, training 3DIS methods like Mask3D, ISBNet, Softgroup, etc., without class supervision is quite challenging, especially given their Backbone + Decoder designs. Typically, the backbone is trained with semantic supervision to support segmentation, and then it’s further trained (or concurrently) with an instance decoder to jointly produce 3D proposals with class IDs.

Could you explain how you managed to train Mask3D without class supervision? As the 3D backbone generally requires prior semantic knowledge, I’m curious if you have any guidance on training the Mask3D backbone using only mask supervision.  As I see, in Mink Backbone, there are layers for predicting semantic segmentation [link.](https://github.com/CVRP-SOLE/SOLE/blob/main/conf/config_base_instance_segmentation.yaml#L21) and [this](https://github.com/CVRP-SOLE/SOLE/blob/main/conf/model/sole.yaml#L54). It produces semantic logits for 200 classes and a set of feature maps at [here](https://github.com/CVRP-SOLE/SOLE/blob/main/models/res16unet.py#L314) so I guess it has already pretrained on ScanNet200.

I’ve attempted training the full Mask3D model without class supervision, but the results seem to have worsened. If I’m misunderstanding any part of the process, please feel free to correct me.

Thank you again for your time and assistance!

PhucNDA.

**Comment by 0nandon** [comment]
I think you can refer to the OpenIns3D paper, since they also trained the Mask3D with only using mask annotations, and we also followed their scheme. Specifically, you should do slight modification to Mask3D, discarding the semantic-related parts, such as :
1. Removing classification head.
2. Removing classification logit in the Hungarian matching : you should only consider logits from mask loss to match the predictions and gt proposals in the training.

For the links you mentioned above:
1. This [link](https://github.com/CVRP-SOLE/SOLE/blob/main/conf/model/sole.yaml#L54) is actually not used neither in the training nor evaluation. If you check the `res16unet.py` file, you may see that `out_channels` argument  is used for initializing the `self.final()` function. However, this `self.final()` function is actually not used in the `forward()` function. So you can just ignore that part.
2. This [link](https://github.com/CVRP-SOLE/SOLE/blob/main/conf/config_base_instance_segmentation.yaml#L21) is for the evaluation. Since SOLE didn't utilize GT class labels, we don't need class information in the training phase.

When we trained Mask3D with only using mask annotations to extract the backbone weight, we didn't use the code in this repository. Instead, we modify the original Mask3D code with referring the setting of OpenIns3D paper. This SOLE repository is for the open-vocabulary training stated in the main paper with leveraging the multi-modal associations and multi-modal framework.

Hope this reply can make it all clear. 

Best,

Seungjun Lee.

**Comment by PhucNDA** [comment]
Hi @0nandon ,

It's all clear.
Thank you for your thoughtful support!
I'm looking forward to using SOLE in future development.

Best,
PhucNDA.
